duckdb_append_blob error without error message

### What happens?

The blob data is 

```sh
[ 10,2,51,57,82,10,158,8,185,41,165,3,180,61,184,18,88,232,132,1,104,251,1,] 
```



### To Reproduce

call with `duckdb_append_blob(pConn.order_appender, ptr, size);` return 1,  

but `duckdb_appender_error(pConn.order_appender),` return null


### OS:

linux

### DuckDB Version:

v0.4.0

### DuckDB Client:

c api

### Full Name:

calvin

### Affiliation:

self

### Have you tried this on the latest `master` branch?

not yet, can not build it

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

```cc
duckdb_state duckdb_append_int32_array(duckdb_appender appender, const int *data, idx_t length) {
    if( length == 0 ) {
        auto empty_list_value = duckdb::Value::EMPTYLIST(duckdb::LogicalType::LIST(duckdb::LogicalType::INTEGER));
        return duckdb_append_internal<duckdb::Value>(appender, empty_list_value);
    }
    duckdb::vector<int> values;
    for(int i = 0; i < length; i++) {
        values.push_back(data[i]);
    }
    auto list_value = duckdb::Value::LIST(duckdb::LogicalType::LIST(duckdb::LogicalType::INTEGER), move(values));
    return duckdb_append_internal<duckdb::Value>(appender, list_value);
}
```

any tips to fix this error:
```sh
In file included from src/main/capi/ub_duckdb_main_capi.cpp:2:
src/main/capi/appender-c.cpp:204:20: error: no matching function for call to 'LIST'
        auto list_value = duckdb::Value::LIST(duckdb::LogicalType::LIST(duckdb::LogicalType::INTEGER), values);
                          ^~~~~~~~~~~~~~~~~~~
src/include/duckdb/common/types/value.hpp:147:26: note: candidate function not viable: no known conversion from 'vector<int>' to 'vector<duckdb::Value>' for 2nd argument
        DUCKDB_API static Value LIST(LogicalType child_type, vector<Value> values);
                                ^
src/include/duckdb/common/types/value.hpp:145:26: note: candidate function not viable: requires single argument 'values', but 2 arguments were provided
        DUCKDB_API static Value LIST(vector<Value> values);
```

--- a/src/main/capi/appender-c.cpp
+++ b/src/main/capi/appender-c.cpp
@@ -95,6 +95,9 @@ duckdb_state duckdb_append_internal(duckdb_appender appender, T value) {
 	auto *appender_instance = (AppenderWrapper *)appender;
 	try {
 		appender_instance->appender->Append<T>(value);
+	} catch (std::exception &ex) {
+		appender_instance->error = ex.what();
+		return DuckDBError;
 	} catch (...) {
 		return DuckDBError;
 	}

--- a/test/api/capi/test_capi_appender.cpp
+++ b/test/api/capi/test_capi_appender.cpp
@@ -405,6 +405,7 @@ TEST_CASE("Test append timestamp in C API", "[capi]") {
 	REQUIRE(status == DuckDBSuccess);
 	REQUIRE(duckdb_appender_error(appender) == nullptr);
 
+	// successful append
 	status = duckdb_appender_begin_row(appender);
 	REQUIRE(status == DuckDBSuccess);
 
@@ -414,6 +415,17 @@ TEST_CASE("Test append timestamp in C API", "[capi]") {
 
 	status = duckdb_appender_end_row(appender);
 	REQUIRE(status == DuckDBSuccess);
+
+	// append failure
+	status = duckdb_appender_begin_row(appender);
+	REQUIRE(status == DuckDBSuccess);
+
+	status = duckdb_append_varchar(appender, "XXXXX");
+	REQUIRE(status == DuckDBError);
+	REQUIRE(duckdb_appender_error(appender) != nullptr);
+
+	status = duckdb_appender_end_row(appender);
+	REQUIRE(status == DuckDBError);
 
 	status = duckdb_appender_flush(appender);
 	REQUIRE(status == DuckDBSuccess);

